src/BoundaryConditions/fill_halo_regions_open.jl

```diff
@@ -1,39 +1,107 @@
-# The syntax for `getbc` is also different for OpenBoundaryCondition than for others,
-# because the boundary-normal index can vary (and array boundary conditions need to be
-# 3D in general).
-
-@kernel function set_west_or_east_u!(u, i_boundary, bc, grid, args) 
+#=function fill_boundary_normal_velocities!(velocities, clock, fields)
+    grid = velocities.u.grid
+
+    fill_open_boundary_regions!(velocities.u, grid, instantiated_location(velocities.u), clock, fields)
+    fill_open_boundary_regions!(velocities.v, grid, instantiated_location(velocities.v), clock, fields)
+    fill_open_boundary_regions!(velocities.w, grid, instantiated_location(velocities.w), clock, fields)
+
+    return nothing
+end=#
+
+const BoundedGrid = Union{AbstractGrid{<:Any, <:Bounded}, 
+                          AbstractGrid{<:Any, <:Any, <:Bounded}, 
+                          AbstractGrid{<:Any, <:Any, <:Any, <:Bounded}}
+
+@inline fill_open_boundary_regions!(field, args...) = fill_open_boundary_regions!(field, field.grid, instantiated_location(field))
+
+@inline fill_open_boundary_regions!(field, grid::BoundedGrid, loc, args...) = fill_open_boundary_regions!(field, field.boundary_conditions, loc, grid, args...)
+
+function fill_open_boundary_regions!(field, boundary_conditions, loc, grid, args...; kwargs...)
+    arch = architecture(grid)
+
+    left_bc = left_boundary_condition(boundary_conditions, loc)
+    right_bc = right_boundary_condition(boundary_conditions, loc)
+
+    launch!(arch, grid, side_plan(loc), fill_open_halo!(loc), field, left_bc, right_bc, loc, grid, args)
+
+    return nothing
+end
+
+fill_open_boundary_regions!(fields::NTuple, boundary_conditions, loc, grid, args...; kwargs...) =
+    [fill_open_boundary_regions!(field, boundary_conditions[n], loc[n], grid, args...; kwargs...) for (n, field) in enumerate(fields)]
+
+@inline left_boundary_condition(boundary_conditions, loc) = nothing
+@inline left_boundary_condition(boundary_conditions, loc::Tuple{Face, Center, Center}) = boundary_conditions.west
+@inline left_boundary_condition(boundary_conditions, loc::Tuple{Center, Face, Center}) = boundary_conditions.south
+@inline left_boundary_condition(boundary_conditions, loc::Tuple{Center, Center, Face}) = boundary_conditions.bottom
+
+@inline right_boundary_condition(boundary_conditions, loc) = nothing
+@inline right_boundary_condition(boundary_conditions, loc::Tuple{Face, Center, Center}) = boundary_conditions.east
+@inline right_boundary_condition(boundary_conditions, loc::Tuple{Center, Face, Center}) = boundary_conditions.north
+@inline right_boundary_condition(boundary_conditions, loc::Tuple{Center, Center, Face}) = boundary_conditions.top
+
+@inline side_plan(loc) = :xy
+@inline side_plan(loc::Tuple{Face, Center, Center}) = :yz
+@inline side_plan(loc::Tuple{Center, Face, Center}) = :xz
+@inline side_plan(loc::Tuple{Center, Center, Face}) = :xy
+
+@inline fill_open_halo!(loc) = _no_fill!
```

Comment:
```suggestion
@inline fill_open_halo!(loc) = nothing
```
will this not work?

Reply:
This is acting like `fill_halos! returned from `permute_boundary_conditions` and (when I tried) you can't do `nothing(whatever...)` so I had to make a do nothing function.